File: app/src/main/java/com/xomena/cmpfutboltfe/RouteStep.java
package com.xomena.cmpfutboltfe;

import android.text.Spanned;

import java.util.ArrayList;
import java.util.List;

public class RouteStep {
    private Spanned stepText;

    public RouteStep(Spanned txt) {
        stepText = txt;
    }

    public Spanned getStepText() {
        return stepText;
    }

    public static List<RouteStep> createRouteStepsList(Spanned[] data) {
        List<RouteStep> steps = new ArrayList<>();

        for (int i = 0; i < data.length; i++) {
            steps.add(new RouteStep(data[i]));
        }

        return steps;
    }
}

File: app/src/main/java/com/xomena/cmpfutboltfe/model/RouteStep.kt
package com.xomena.cmpfutboltfe.model

import android.text.Spanned

data class RouteStep(
    val stepText: Spanned
) {
    companion object {
        @JvmStatic
        fun createRouteStepsList(data: Array<Spanned>): List<RouteStep> {
            return data.map { RouteStep(it) }
        }
    }
}
